src/arch/aarch64/smc.c

```diff
@@ -109,10 +133,44 @@ bool handle_smc(size_t vcpu_id, uint32_t hsr)
             }
             LOG_VMM_ERR("Unhandled SMC: standard service call %lu\n", fn_number);
             break;
+#if defined(CONFIG_ALLOW_SMC_CALLS)
+        case SMC_CALL_SIP_SERVICE:
+            return handle_sip(vcpu_id, &regs, fn_number);
+#endif
         default:
             LOG_VMM_ERR("Unhandled SMC: unknown value service: 0x%lx, function number: 0x%lx\n", service, fn_number);
             break;
     }
 
     return false;
 }
+
+
+/* Default handler of calls to a Service (group of Functions) running at Secure Monitor level;
+  forwards all the calls without applying any policy
+*/
+bool default_handle_service(uint64_t vcpu_id, seL4_UserContext *regs, uint64_t fn_number)
+{
+
+    seL4_CPtr smc_cap = SMC_CAP;
```

Comment:
While it isn't the current case for the rest of the library, I would like the library to be agnostic to the seL4 environment it is in, this means that it should be up to the user to provide the library with what capability to use for SMC calls.

If it's not too hard for you, it'd be good if specifying the SMC cap was part of the API.

Reply:
Right, good catch, I'll change it.

Reply:
Done